From 075bac13cdb53554df651056d15d0c26f95fcab2 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 14 Apr 2018 10:52:50 +0300
Subject: [PATCH] Add dsv support

Tool that will be rum from U-Boot for DDR validation
---

diff --git a/Config.in b/Config.in
index 14d49a3..cca2b34 100644
--- a/Config.in
+++ b/Config.in
@@ -4,5 +4,6 @@
 source "$BR2_EXTERNAL_HL_DALI_PATH/package/specs_external/Config.in"
 source "$BR2_EXTERNAL_HL_DALI_PATH/package/uboot-test-hooks/Config.in"
 source "$BR2_EXTERNAL_HL_DALI_PATH/package/tbot/Config.in"
+source "$BR2_EXTERNAL_HL_DALI_PATH/package/dsv/Config.in"
 
 # Below can add to make it appear in the top-level configuration menu
diff --git a/Makefile b/Makefile
index f446bf9..ae9e3f2 100644
--- a/Makefile
+++ b/Makefile
@@ -58,6 +58,7 @@
 UBOOT_BASE := ${OUTPUT_DIR}/build/uboot-custom
 LINUX_BASE := ${OUTPUT_DIR}/build/linux-custom
 UBOOT_XMPL := ${UBOOT_BASE}/examples/standalone
+DSV_BASE := ${OUTPUT_DIR}/build/dsv-custom
 .PHONY: dali-onix-install dali-pldm-install dali-asic-install
 dali-onix-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
 dali-onix-install: DALI_INSTALL_HOST_IP ?= monster4
@@ -78,6 +79,7 @@
 	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp -r $(UBOOT_XMPL)/dali_*.bin        root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 dali-onix-install dali-pldm-install dali-asic-install: dali-install
 dali-onix-install:
 	@
diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index 459e8cc..0c4c11c 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -6,3 +6,4 @@
 SPECS_EXTERNAL_OVERRIDE_SRCDIR = $(TOPDIR)/../specs_external
 UBOOT_TEST_HOOKS_OVERRIDE_SRCDIR = $(TOPDIR)/../uboot-test-hooks
 TBOT_OVERRIDE_SRCDIR = $(TOPDIR)/../tbot
+DSV_OVERRIDE_SRCDIR = $(TOPDIR)/../dsv
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index 482d095..300a3ef 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -61,3 +61,4 @@
 BR2_PACKAGE_ARMCP=y
 BR2_PACKAGE_UBOOT_TEST_HOOKS=y
 BR2_PACKAGE_TBOT=y
+BR2_PACKAGE_DSV=y
diff --git a/package/dsv/Config.in b/package/dsv/Config.in
new file mode 100644
index 0000000..9557e2e
--- /dev/null
+++ b/package/dsv/Config.in
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_DSV
+	bool "DSV"
+	help
+	  Dram System Validation (DSV) is a tool that will run from U-Boot
diff --git a/package/dsv/dsv.mk b/package/dsv/dsv.mk
new file mode 100644
index 0000000..d46bfb3
--- /dev/null
+++ b/package/dsv/dsv.mk
@@ -0,0 +1,9 @@
+DSV_VERSION = 1.0.0
+DSV_SITE = dsv
+DSV_LICENSE = unclear
+
+define DSV_BUILD_CMDS
+	$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all
+endef
+
+$(eval $(generic-package))
